spatial-tracking-explainer.md

```diff
@@ -0,0 +1,382 @@
+
+Other XR hardware that is orientation-only tracking, may provide an emulated value for the floor offset.  On these devices, it is recommended that the User Agent or underlying platform provide a setting for users to customize this value.
+
+### Transitioning between XRFrameOfReferences
+It is expected that developers will often choose to preview `immersive` experiences with a similar experience `inline`.  In this situation, users often expect to see the scene from the same perspective when they make the transition from `inline` to `immersive`. In most cases, developers can accomplish this by computing the transform between the user and the virtual scene's origin in the previous `XRSession`.  When the new `XRSession` is initialized, the same transform can be applied to the virtual scene's origin in the new `XRFrameOfReference`.
+
+### Handling a Tracking System Reset
+Many XR systems have a mechanism for allowing the user to reset which direction is "forward" or re-center the scene's origin at their current location. For security and comfort reasons the WebXR Device API has no mechanism to trigger a pose reset programmatically, but it can still be useful to know when it happens. Pages may want to take advantage of the visual discontinuity to reposition the user or other elements in the scene into a more natural position for the new orientation. Pages may also want to use the opportunity to clear or reset any additional transforms that have been applied if no longer needed.
+
+The exact behavior of a reset is platform specific, but in most cases it's expected that the origin and/or forward direction of the `XRStationaryFrameOfReference` will shift to be aligned to the users current physical location and orientation. On `XRBoundedFrameOfReference`, it is also expected that the `boundsGeometry` may have changed.
+
+A page can be notified when a pose reset happens by listening for the 'onreset' event from the `XRStationaryFrameOfReference` or `XRBoundedFrameOfReference`.  The `XRUnboundedFrameOfReference` will not receive this event.  The event must fire prior to any poses being delivered with the new origin/direction, and all poses queried following the event must be relative to the reset origin/direction. 
+
+```js
+xrFrameOfReference.addEventListener('reset', xrFrameOfReferenceEvent => {
+  // For an app that allows artificial Yaw rotation, this would be a perfect
+  // time to reset that.
+  resetYawTransform();
+
+  // For an app using the XRBoundedFrameOfReference, this would be a perfect time to
+  // reset the bounds and potentially re-layout content intended to be reachable within
+  // the bounds
+  createBoundsMesh();
+});
```

Comment:
In the spirit of "onReset", should there be an event available on the FoR to tell programmers the state of the tracking system?  So, if the tracking system is temporarily in a bad state, the app can react reasonably?

Reply:
Yeah, originally I'd been hoping to merge this PR without needing to address tracking loss first (it's in the list of post-merge topics in the PR description).  The reason is that it's somewhat a thorny problem when you have to take into consideration all the different hardware behaviors.  However, I'm sufficiently convinced that the solution is too fundamental to do after-the-fact.  Poking at it now.

Reply:
Ok, I've been talked into (and reasonably so) dividing the question of tracking loss handling in half.  In this PR I'll be updating the onreset to fire when a tracking loss recovery causes a disconnected map fragment.  However, there is still a general question of WebXR's behavior *during* tracking loss.  I'll file that as a separate issue as originally planned.